pyomo/contrib/mindtpy/iterate.py

```diff
@@ -299,6 +286,7 @@ def algorithm_should_terminate(solve_data, config, check_cycling):
         return True
 
     # Check if algorithm is stalling
+    # BUG TODO: here only applies to minimization problems.
```

Comment:
Oh, David. Actually, I forgot the meaning of stalling. Does it mean the primal bound or the dual bound not making progress? I checked DICOPT. There is one related option.

> fp_stalllimit (integer): Stop when no improving solution has been bound for a number of FP iterations.

Reply:
Yes, stalling means that not enough progress has been made in several iterations. For feaspump, if after a certain number of iterations there was no progress using fp, we would turn on OA.

Reply:
> Yes, stalling means that not enough progress has been made in several iterations.

What does progress mean here? Primal bound progress or dual bound progress?

Reply:
Given that it was for the feasibility pump, it was the primal bound. In this context it could also be the primal bound (no better solution was found after X iterations)